src/toolbox/shamir_secrets.rs

```diff
@@ -148,7 +151,26 @@ impl SecretShare {
     ///
     /// The sparse_shares vector should follow the "index + 1 = x" convention, but is allowed to contain Option values
     /// to represent missing shares.  DO NOT include the secret in sparse_shares, as it is passed in separately.
+    /// 
+    /// Corner cases:  need sanity checks on threshold and size of sparse_shares
+    /// The size of the vector we return is going to need to match the size of sparse_shares.  Do our shares match the is_some indices, or the is_none indices?
     pub fn complete<R: CryptoRng + RngCore>(secret: Scalar, threshold: usize, sparse_shares: &Vec<Option<Scalar>>, rng: &mut R) -> Result<SecretShare, String> {
+        // println!("Asked to complete SSS with threshold {} and sparse_shares {:?}", threshold, sparse_shares);
+        if threshold == 0 {
+            // Threshold of 0 means there are no unknowns in our proof.  In that case, we share just the "secret" itself
+            return Ok(SecretShare {
+                secret,
+                shares: vec![Some(secret); sparse_shares.len()],
+                threshold,
+            })
+        } else if threshold == 1 {
+            // the only possible share is the secret itself; I think we hand back nothing?
+            return Ok(SecretShare {
+                secret,
+                shares: sparse_shares.clone(),
+                threshold,
+            })
+        }
```

Comment:
This part is wrong, or at least, not well thought out.  Still need to ponder on what to do in these corner cases.